src/visGeometry/ColorHandler.ts

```diff
@@ -0,0 +1,201 @@
+import { map, round, isEqual } from "lodash";
+import { Color } from "three";
+import { AgentColorInfo } from "./types";
+
+export function convertColorStringToNumber(color: number | string): number {
+    if (typeof color !== "string") {
+        return color;
+    }
+    return parseInt(color.toString().replace(/^#/, "0x"), 16);
+}
+
+export function convertColorNumberToString(color: number | string): string {
+    if (typeof color === "string") {
+        return color;
+    }
+    return "#" + new Color(color).getHexString();
+}
+
+export const checkHexColor = (color: string): string => {
+    const hexColorCodeRegex = /^#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$/;
+    if (!hexColorCodeRegex.test(color)) {
+        throw new Error("Invalid color code");
+    }
+    return color;
+};
+
+class ColorHandler {
+    public idColorMapping: Map<number, number>;
+    private colorsData: Float32Array;
+
+    constructor() {
+        this.idColorMapping = new Map<number, number>();
+        // will be set by setColorArray, but need to initialize
+        // so that typescript doesn't complain
+        this.colorsData = new Float32Array(0);
+    }
+
+    private get numberOfColors(): number {
+        return this.colorsData.length / 4;
+    }
+
+    private setColorArray(colors: (number | string)[]): Float32Array {
+        const colorNumbers = colors.map(convertColorStringToNumber);
+        const numberOfColors = colors.length;
+        // fill buffer of colors:
+        const colorsData = new Float32Array(numberOfColors * 4);
+        for (let i = 0; i < numberOfColors; i += 1) {
+            // each color is currently a hex value:
+            colorsData[i * 4 + 0] =
+                ((colorNumbers[i] & 0x00ff0000) >> 16) / 255.0;
+            colorsData[i * 4 + 1] =
+                ((colorNumbers[i] & 0x0000ff00) >> 8) / 255.0;
+            colorsData[i * 4 + 2] =
+                ((colorNumbers[i] & 0x000000ff) >> 0) / 255.0;
+            colorsData[i * 4 + 3] = 1.0;
+        }
+        return colorsData;
+    }
```

Comment:
Should this be `makeColorDataArray`? `set` implies it's setting an internal value, which it doesn't seem like it's doing.